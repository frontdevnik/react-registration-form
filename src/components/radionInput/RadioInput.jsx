import React from 'react';

export default ({ name, text = '', register, checked }) => {
  return (
    <label>
      <input
        name={name}
        type="radio"
        ref={register()}
        checked={checked}
        value={text}
      />
      <span>{text}</span>
    </label>
  );
};
