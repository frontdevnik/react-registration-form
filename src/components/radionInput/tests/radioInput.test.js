import React from 'react';

import RadioInput from '../RadioInput';

import { radioInputProps } from './stubProps';

describe('<RadioInput />', () => {
  const component = shallow(<RadioInput {...radioInputProps} />);

  it('should render correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });

  it('should input name is correct', () => {
    expect(component.find('input').props().name).toBe(radioInputProps.name);
  });

  it('should input type is correct', () => {
    expect(component.find('input').props().type).toBe('radio');
  });

  it('should render span with correct text', () => {
    expect(component.find('span').text()).toBe(radioInputProps.text);
  });
});
