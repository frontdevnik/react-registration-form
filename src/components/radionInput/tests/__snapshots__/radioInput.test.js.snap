// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioInput /> should render correctly 1`] = `"<label><input type=\\"radio\\" name=\\"name\\" value=\\"text\\"/><span>text</span></label>"`;
